### PR TITLE
Align hashing with 'canonical' implementation for GNAT

### PR DESCRIPTION
**Pull request by dzajkowski**
## Overview
<sup>_What this PR does, and why it's needed_</sup>
The previous implementation had a 'canonical' representation for GNAT.
This change aligns the new implementation with the previous assumptions.

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>
https://rchain.atlassian.net/browse/RCHAIN-3309

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: rspace/src/test/scala/coop/rchain/rspace/nextgenrspace/history/EncodingSpec.scala [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  "WaitingContinuation list encode" should "return same hash for different orderings of each continuation" in forAll {
+    (c1: Continuation, c2: Continuation, c3: Continuation) =>
+      val bytes1 = HistoryRepositoryImpl.encodeContinuations(c1 :: c2 :: c3 :: Nil)
+      val bytes2 = HistoryRepositoryImpl.encodeContinuations(c3 :: c2 :: c1 :: Nil)
+      val bytes3 = HistoryRepositoryImpl.encodeContinuations(c2 :: c3 :: c1 :: Nil)
+      Blake2b256Hash.create(bytes1) shouldBe Blake2b256Hash.create(bytes2)
+      Blake2b256Hash.create(bytes1) shouldBe Blake2b256Hash.create(bytes3)
+  }
+
+  "Joins list encode" should "return n same hash for different orderings of each channel list" in forAll {
```
Comment (lukasz-golebiewski) [inline rspace/src/test/scala/coop/rchain/rspace/nextgenrspace/history/EncodingSpec.scala]: Typo

**Comment by lukasz-golebiewski** [comment]
bors retry

**Comment by lukasz-golebiewski** [comment]
bors try

File: shared/src/test/scala/coop/rchain/shared/GeneratorUtils.scala [hunk 2]
```diff
@@ -13,17 +13,21 @@ object GeneratorUtils {
   def distinctListOf[T: Arbitrary] =
     distinctListOfGen(arbitrary[T])(_ == _)
 
+  def limitedDistinctList[T: Arbitrary](maxSize: Int) =
```
Comment (lukasz-golebiewski) [inline shared/src/test/scala/coop/rchain/shared/GeneratorUtils.scala]: This shouldn't be needed. It should be enough to supply the `size` parameter to the `forAll` construct

**Comment by dzajkowski** [comment]
bors try

**Comment by dzajkowski** [comment]
bors try

**Comment by dzajkowski** [comment]
bors try

**Comment by dzajkowski** [comment]
bors try

**Comment by dzajkowski** [comment]
bors try-

**Comment by dzajkowski** [comment]
bors try

**Comment by dzajkowski** [comment]
bors try

**Pull request merged**